orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -361,8 +361,14 @@ def forward(ctx, *inputs, **kwargs):
                 _create_iobinding(self._training_io_binding, inputs, self._onnx_training, self._device)
 
                 # Run and return module outputs.
+                run_id = []
                 user_outputs = tuple(_ort_output_to_torch_tensor(forward_output) \
-                    for forward_output in self._training_session.run_forward(self._training_io_binding, self._run_options))
+                    for forward_output in self._training_session.run_forward(self._training_io_binding, self._run_options, run_id))
```

Comment:
If only a single entry is expected in ```run_id```, may we return a single number (```int64_t``` from ```run_forward```) instead of populating the list with a single entry?

Reply:
run_id returned in a tuple now.

Reply:
This is a trick to make python binding work with c++.
Python always passes int (any primitive type) by value whereas lists are passed by reference. An alternative would be make this API to return a tuple with run_id on it as opposed to pass it as argument to the method

Reply:
Sorry, I don't follow. It appears that ```run_id``` is defined as a list here and then passed into ```run_forward``` to get populated. I am asking if it is possible for ```run_forward``` to return a tuple (with ```forward_outputs``` and ```run_id```). I think @thiagocrepaldi is also suggesting the same thing (if I am not mistaken).